cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/dataset/MetadataDataset.java

```diff
@@ -31,7 +31,9 @@
 import co.cask.cdap.data2.metadata.indexer.Indexer;
```

Comment:
Make sure you understand the serialization of metadata, to think through whether these changes make it backwards compatible.
(i.e. is it the bytes of the toString() of the Id that is used during serialization?)